instrumentation/kafka/kafka-connect-2.6/testing/src/test/java/io/opentelemetry/instrumentation/kafkaconnect/v2_6/MongoKafkaConnectSinkTaskTest.java

```diff
@@ -0,0 +1,728 @@
+            .withEnv("CONNECT_REST_PORT", String.valueOf(CONNECT_REST_PORT_INTERNAL))
+            .withEnv("CONNECT_GROUP_ID", "kafka-connect-group")
+            .withEnv("CONNECT_CONFIG_STORAGE_TOPIC", "kafka-connect-configs")
+            .withEnv("CONNECT_OFFSET_STORAGE_TOPIC", "kafka-connect-offsets")
+            .withEnv("CONNECT_STATUS_STORAGE_TOPIC", "kafka-connect-status")
+            .withEnv("CONNECT_KEY_CONVERTER", "org.apache.kafka.connect.json.JsonConverter")
+            .withEnv("CONNECT_VALUE_CONVERTER", "org.apache.kafka.connect.json.JsonConverter")
+            .withEnv("CONNECT_CONFIG_STORAGE_REPLICATION_FACTOR", "1")
+            .withEnv("CONNECT_OFFSET_STORAGE_REPLICATION_FACTOR", "1")
+            .withEnv("CONNECT_STATUS_STORAGE_REPLICATION_FACTOR", "1")
+            .waitingFor(
+                Wait.forHttp("/")
+                    .forPort(CONNECT_REST_PORT_INTERNAL)
+                    .withStartupTimeout(Duration.of(5, MINUTES)))
+            .withStartupTimeout(Duration.of(5, MINUTES))
+            .withCommand(
+                "bash",
+                "-c",
+                "mkdir -p /var/log/kafka-connect && "
+                    + "confluent-hub install --no-prompt --component-dir /usr/share/java "
+                    + "mongodb/kafka-connect-mongodb:1.11.0 && "
+                    + "echo 'Starting Kafka Connect with logging to /var/log/kafka-connect/' && "
+                    + "/etc/confluent/docker/run 2>&1 | tee /var/log/kafka-connect/kafka-connect.log");
+
+    mongoDB =
+        new MongoDBContainer(DockerImageName.parse("mongo:4.4"))
+            .withNetwork(network)
+            .withNetworkAliases(MONGO_NETWORK_ALIAS)
+            .withStartupTimeout(Duration.of(5, MINUTES));
+
+    // Start containers (backend already started)
+    Startables.deepStart(Stream.of(zookeeper, kafka, kafkaConnect, mongoDB)).join();
+
+    // Wait until Kafka Connect container is ready
+    given()
+        .contentType(ContentType.JSON)
+        .when()
+        .get(getKafkaConnectUrl())
+        .then()
+        .statusCode(HttpStatus.SC_OK);
+  }
+
+  @BeforeEach
+  public void reset() {
+    deleteConnectorIfExists();
+    clearMongoCollection();
+    // Clear spans from backend (like smoke tests)
+    clearBackendTraces();
+  }
+
+  private static void clearBackendTraces() {
+    try {
+      String backendUrl = getBackendUrl();
+      given().when().get(backendUrl + "/clear").then().statusCode(200);
+    } catch (RuntimeException e) {
+      // Ignore failures to clear traces
+    }
+  }
+
+  private static String getBackendUrl() {
+    return format(
+        Locale.ROOT, "http://%s:%d", backend.getHost(), backend.getMappedPort(BACKEND_PORT));
+  }
+
+  @Test
+  public void testKafkaConnectMongoSinkTaskInstrumentation()
+      throws IOException, InterruptedException {
+    // Create unique topic name
+    String uniqueTopicName = TOPIC_NAME + "-" + System.currentTimeMillis();
```

Comment:
It is best not to do this but rather use the `TOPIC_NAME `. Messaging semantic conventions require that span names have low cardinality https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/#span-name. Here you make the topic name look like it was a temporary name so whoever is looking at it will think that there must be a bug in the instrumentation because such name shouldn't be captured in span name.